websites/R/Radio.net/presence.ts

```diff
@@ -1,187 +1,173 @@
 const presence = new Presence({
-    clientId: "634124614544392193"
-  }),
-  strings = presence.getStrings({
-    play: "presence.playback.playing",
-    pause: "presence.playback.paused",
-    search: "presence.activity.searching",
-    browsing: "presence.activity.browsing"
-  });
+  clientId: "634124614544392193"
+});
 
-let lastPath = "", //Last played radio station or podcast
-  browsingTimestamp = 0; //Timestamp when started listening to a radio station
+let strings: Awaited<ReturnType<typeof getStrings>>, timestamp: number;
 
 presence.on("UpdateData", async () => {
-  //Current path
-  const path = window.location.pathname.split("/").slice(1),
-    //Presence data
-    presenceData: PresenceData = {
-      details: "Radio.net",
-      largeImageKey: "logo_big"
-    },
-    //Document title
-    { title } = document,
-    //Document header
-    header = document.querySelector("h1") as HTMLElement;
-
-  switch (path[0]) {
-    //Radio
-    //Podcast
-    case "s":
-    case "p": {
-      if (path[1] !== lastPath || browsingTimestamp === 0) {
-        browsingTimestamp = Math.round(Date.now() / 1000);
-        [, lastPath] = path;
-      }
-
-      //Player State
-      const status = document.querySelector<HTMLElement>(".player__info-wrap");
-
-      if (
-        document.querySelector<HTMLElement>(".player__animate-icon").style
-          .display !== "none"
-      ) {
-        //Playing
-        presenceData.details = header.textContent;
+  const host = window.location.hostname.split("."),
+    path = window.location.pathname.split("/").slice(1),
+    presenceData: PresenceData = {};
+
+  strings = await getStrings();
```

Comment:
You should only fetch it once and if the language got changed. useless to fetch it everytime on updatedata